packages/stargate/src/modules/staking/aminomessages.ts

```diff
@@ -29,10 +29,10 @@ interface Description {
   readonly details: string;
 }
 
-function protoDecimalToJson(decimal: string): string {
+export function protoDecimalToJson(decimal: string): string {
   const parsed = Decimal.fromAtomics(decimal, 18);
   const [whole, fractional] = parsed.toString().split(".");
-  return `${whole}.${fractional.padEnd(18, "0")}`;
+  return `${whole}.${(typeof fractional !== "undefined" ? fractional : "").padEnd(18, "0")}`;
```

Comment:
This is the same as your code in modern notation:

```suggestion
  return `${whole}.${(fractional ?? "").padEnd(18, "0")}`;
```

Reply:
Zero is considered false, hence the use of typeof.

Reply:
Hum in fact "0" is considered true, it was the initial absence of parenthesis that generated an error. In your proposal the brackets are kept and it works.

Reply:
The `??` operator only uses the fallback when the value is null or undefined. So even `0 ?? "fallback"` and `"" ?? "fallback"` preserve the original value. This is different from using `||`.